creator-node/src/snapbackSM/secondarySyncHealthTracker.js

```diff
@@ -131,27 +138,35 @@ const Setters = {
 
 const Getters = {
   /**
-   * Given wallet and secondaries array, returns map from each secondary to successCount, failureCount, and successRate
+   * Given a mapping of wallet to secondaries arrays, returns mapping from wallet to
+   * sync metrics for that wallet, where sync metrics are a mapping of secondary endpoint
+   * to successCount, failureCount, and successRate
    *
-   * @param {String} wallet
-   * @param {Array} secondaries
-   * @returns {Object} { secondary1: { 'successCount' : _, 'failureCount': _, 'successRate': _ }, ... }
+   * @param {Object { <wallet (string)>: <secondary endpoints (string array)}} walletsToSecondariesMapping
+   * @returns {Object} { '0x...': { 'https://secondary1...': { 'successCount' : _, 'failureCount': _, 'successRate': _ }, ... } ... }
    */
-  async computeUserSecondarySyncSuccessRates(wallet, secondaries) {
+  async computeUsersSecondarySyncSuccessRates(walletsToSecondariesMapping) {
     // Initialize sync success and failure counts for every secondary to 0
-    const secondarySyncMetrics = {}
-    secondaries.forEach((secondary) => {
-      secondarySyncMetrics[secondary] = { successCount: 0, failureCount: 0 }
-    })
+    const secondarySyncMetricsMap = {}
+    const wallets = Object.keys(walletsToSecondariesMapping)
+    for (const wallet of wallets) {
+      const secondarySyncMetrics = secondarySyncMetricsMap[wallet] || {}
+      for (const secondary of walletsToSecondariesMapping[wallet]) {
+        secondarySyncMetrics[secondary] = { successCount: 0, failureCount: 0 }
+        secondarySyncMetricsMap[wallet] = secondarySyncMetrics
+      }
+    }
 
-    // Retrieve map of all SyncRequestOutcome keys and daily counts for user from all secondaries
+    // Retrieve map of all SyncRequestOutcome keys and daily counts for wallets from all secondaries
     const userSecondarySyncHealthOutcomes =
-      await Getters.getSyncRequestOutcomeMetrics({ wallet })
+      await Getters.batchGetSyncRequestOutcomeMetrics(wallets)
```

Comment:
took me a while to figure out that this is only getting the metrics for today, bc if you don't pass in a `date` to `const pattern = Utils._getRedisKeyPattern({})` it only gets you data for today

on me for not documenting earlier, mind adding / clarifying? maybe we can change function names where appropriate as well

Reply:
sure thing! pushed some improvements to this